File: Excel/444 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/443 Look and Say Sequence.xlsx", range = "A1:A10")
test  = read_excel("Excel/443 Look and Say Sequence.xlsx", range = "B1:B10")


generate_next = function(number) {
  number_str = as.character(number)
  digits = str_split(number_str, "")[[1]]
  
  unique_digits = unique(digits)
  
  result = map_chr(unique_digits, function(digit) {
    count = sum(digits == digit)
    paste0(count, digit)
  }) %>% paste0(collapse = "")
  
  as.numeric(result)
}

generate_sequence = function(start_digit, iter = 4) {
  result = start_digit
  
  for (i in 1:iter) {
    next_number = generate_next(result[length(result)])
    result = c(result, next_number)
  }
  
  all = result %>%
    setdiff(., start_digit) %>%
    paste0(collapse = ", ")
  
  return(all)
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Numbers, generate_sequence))

identical(result$`Answer Expected`, test$`Answer Expected`)
# [1] TRUE

File: Excel/444 Challenge.py
import pandas as pd

input = pd.read_excel("443 Look and Say Sequence.xlsx", usecols="A", nrows=10)
test = pd.read_excel("443 Look and Say Sequence.xlsx", usecols="B", nrows=10)

def generate_next(number):
    number_str = str(number)
    digits = []
    
    for digit in number_str:
        if digit not in digits:
            digits.append(digit)
    
    result = ""
    for digit in digits:
        count = number_str.count(digit)
        result += str(count) + digit
    
    return int(result)

def generate_sequence(start_digit, iter=4):
    result = [start_digit]
    
    for i in range(iter):
        next_number = generate_next(result[-1])
        result.append(next_number)
    
    all_numbers = [str(num) for num in result if num != start_digit]
    return ", ".join(all_numbers)

result = input.copy()
result["Answer Expected"] = result["Numbers"].map(lambda x: generate_sequence(x))

print(result["Answer Expected"].equals(test["Answer Expected"])) # True